.golangci.yaml

```diff
@@ -0,0 +1,135 @@
+run:
+  timeout: 4m
+
+linters:
+  disable-all: false
+  enable:
+    - gocyclo
+    - misspell
+    - nilerr
+    - unparam
+    - gosec
+    - unused
+    - govet
+    - gosimple
+    - errorlint
+    - errcheck
+    - dogsled
+    - cyclop
+    - exhaustive
+    - nestif
+    - goconst
+    - gofmt
+    - revive
+    - makezero
+    - nakedret
+    - prealloc
+    - nolintlint
+    - staticcheck
+    - thelper
+    - whitespace
+
+linters-settings:
+  gocyclo:
+    min-complexity: 25
```

Comment:
Some of these settings is rather high, but I want to get this PR merged. We can work on lovering these setting at a later stage.